source/develop/release-management/features/storage/incremental-backup.html.md

```diff
@@ -12,173 +12,421 @@ feature_status: Design
-up the current changes in the active snapshot. Another use case may be
-backing up the previous snapshot if a user added a snapshot since the
-last backup.
+Incremental backup allows a user to backup only the changed blocks since
+the last backup, instead of performing a full backup on every backup.
+If the VM is running during the backup, new data written after the
+backup is started is not included in the backup.
 
-Note: users using RAW files will have to create a snapshot on top to
-allow incremental backup to start.
+## Owner
 
+- Nir Soffer (<nsoffer@redhat.com>)
+- Daniel Erez (<derez@redhat.com>)
 
-## Owner
 
-*   Nir Soffer (<nsoffer@redhat.com>)
-*   Daniel Erez (<derez@redhat.com>)
+## High level Design
+
+### Creating VM
+
+When adding a disk, the user should enable incremental backup for every
+disk. If incremental backup is enabled for a disk, a backup application
+can included the disk during incremental backups.
+
+Since incremental backup requires qcow2 format, disks enabled for
+incremental backup will use qcow2 format instead of raw format. See
+[Disk Format](#disk-format) for more info.
+
+Disks not marked for incremental backup can be backed in the same way
+they were backed in the past.
+
+### Enabling existing VM for incremental backup
+
+Since raw disks are not supported, a user needs to create a snapshot
+including the disks to enabled incremental backup for the disks.  This
+creates a qcow2 layer on top of the raw disk, that can be used to
+perform incremental backups from this point.
+
+### Deleting snapshots on existing VMs
+
+If the base layer of a disk is using raw format, deleting the last
+snapshot, merging the top qcow2 layer into the base layer will convert
+the disk to raw, and disable incremental backup. The user can create a
+new snapshot including this disk to enable back incremental backup.
+
+### Disk format
+
+Here is a table showing how enabling incremental backup affects disk
+format.
+
+```
+storage     provisioning        incremental     format
+====================================================================
+block       thin                enabled         qcow2
+block       preallocated        enabled         qcow2 (preallocated)
+file        thin                enabled         qcow2
+file        preallocated        enabled         qcow2 (preallocated)
+--------------------------------------------------------------------
+block       thin                disabled        qcow2
+block       preallocated        disabled        raw (preallocated)
+file        thin                disabled        raw (sparse)
+file        preallocated        disabled        raw (preallocated)
+--------------------------------------------------------------------
+network     -                   disabled        raw
+lun         -                   disabled        raw
+```
+
+### Incremental backup flow
 
+1. Backup application finds VM disks enabled for incremental backup
+   using oVirt API.
```

Comment:
Not sure we need this step, since engine can tell which disk are enabled for incremental backup, so it can create the right request for the caller, trying to do incremental backup for the qcow2 disks, and full backup for the raw disks.